server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java

```diff
@@ -334,15 +334,34 @@ public boolean isInSortOrderExecutionRequired() {
         }
 
         /**
-         * Return false if this aggregation or any of the child aggregations does not support concurrent search
+         * Return false if this aggregation or any of the child aggregations does not support parallel collection.
+         * As a result, a request including such aggregation is always executed sequentially despite concurrency is enabled for the query
+         * phase. Note: aggregations that don't support parallel collection, may or may not support offloading collection to the search
+         * worker threads, depending on what {@link #supportsOffloadingSequentialCollection()} returns.
          */
-        public boolean supportsConcurrentExecution() {
+        // TODO this is not used yet, it will be used once we enable parallel collection for the query phase
+        public boolean supportsParallelCollection() {
             for (AggregationBuilder builder : aggregationBuilders) {
-                if (builder.supportsConcurrentExecution() == false) {
+                if (builder.supportsParallelCollection() == false) {
                     return false;
                 }
             }
-            return isInSortOrderExecutionRequired() == false;
+            return true;
```

Comment:
Should this method also return `supportsOffloadingSequentialCollection()` here instead of `true`? Seems like we should be careful here as now we have two things controlling parallelism. Don't want them getting out of sync.

Reply:
I spent the best part of my day trying to make this work. I don't like the need for two knobs here and it's not straight-forward what they do. My hope is that we can get rid of both soon.

These two methods are really just exposing the value returned by the aggregation builders, and that already includes the fallback logic, so we should be fine. I will though add additional unit tests for this, and verify that the two are aligned.